### PR TITLE
a bunch of fixes to the daily email PR

### PR DESCRIPTION
**Pull request by owocki**
makes a bunch of changes to the new daily email so that itll actually look good in prod:
1. adds the ability to add arbitrary dates (for workshops, livestreams, grant start dates, etc)
2. adds links to activity feed
3. hackathon visibility filter
4. progressively email, starting with 20% of the userbase.. even if they have no bounties.
5. adds founder announcements to th etop
6. speed logs

tested at http://localhost:8000/_administration/email/new_bounty_daily

### CODE REVIEW

**Comment by owocki** [comment]
this is ready for review, would like to get it in if possible to make usre the first new daily email ppl get is good.

note: i cant run the linter due to a few upstream issues @sebastiantf @gitcoinco/engineers 

```

app/assets/v2/css/town_square.css
 1300:1  ✖  Unexpected duplicate selector ".dark-mode .alert-info", first used at line 1281                                 no-duplicate-selectors
 1304:1  ✖  Unexpected duplicate selector ".dark-mode .web3modal-provider-name", first used at line 1285                    no-duplicate-selectors
 1307:1  ✖  Unexpected duplicate selector ".dark-mode .gc-search-box input", first used at line 1288                        no-duplicate-selectors
 1310:1  ✖  Expected selector ".dark-mode .gc-search-result__content p" to come before selector "#get-started p"            no-descending-specificity
 1310:1  ✖  Expected selector ".dark-mode .gc-search-result__content p" to come before selector "#get-started a p"          no-descending-specificity
 1310:1  ✖  Unexpected duplicate selector ".dark-mode .gc-search-result__content p", first used at line 1090                no-duplicate-selectors
 1313:1  ✖  Unexpected duplicate selector ".dark-mode .gc-search-result__content, .dark-mode .gc-search-box, .dark-mode     no-duplicate-selectors
            .web3modal-provider-container, .dark-mode .web3modal-provider-wrapper, .dark-mode .web3modal-modal-card",
            first used at line 1291

```

File: app/marketing/models.py [hunk 1]
```diff
@@ -407,3 +407,21 @@ def get_absolute_url(self):
...
+class UpcomingDate(SuperModel):
+    """Define the upcoming date model"""
+
+    title = models.CharField(max_length=255)
+    date = models.DateTimeField(db_index=True)
+    img_url = models.URLField(db_index=True, blank=True)
+    url = models.URLField(db_index=True)
+    comment = models.TextField(max_length=255, default='', blank=True)
+
+    @property
```
Comment (owocki) [inline app/marketing/models.py:420]: todo in the future, make grants rounds, hackathons, workshops auto create these objects.

File: app/retail/emails.py [hunk 2]
```diff
@@ -552,14 +553,6 @@ def render_new_bounty(to_email, bounties, old_bounties, offset=3, quest_of_the_d
         pass        
 
     upcoming_events = []
-    if upcoming_grant:
-        upcoming_events.append({
-            'event': upcoming_grant,
-            'title': upcoming_grant.title,
-            'image_url': upcoming_grant.logo.url if upcoming_grant.logo else f'{settings.STATIC_URL}v2/images/emails/grants-neg.png',
```
Comment (owocki) [inline app/retail/emails.py:559]: didnt make any sense to show grants by their clr calc date IMHO.  

if we want to include trending grants in the future.. we should do it in its own section. @sebastiantf lmk if you waant to work on that.

Comment (owocki) [reply app/retail/emails.py:559]: @sebastiantf another thing i could use help on is that the email takes about 2s to calculate. happy to write a v2 ticket where you do the above and also profile/improve the performance of it so we can send 30k/day (scaling up to 100k in the next year)

Comment (sebastiantf) [reply app/retail/emails.py:559]: > didnt make any sense to show grants by their clr calc date IMHO.
> 
> if we want to include trending grants in the future.. we should do it in its own section. @sebastiantf lmk if you waant to work on that.

Yeah sure

**Comment by owocki** [comment]
looks like upstream linter issues have been fixed. thanks @octavioamu

File: app/retail/emails.py (hunk 2 above)
Comment (owocki) [reply app/retail/emails.py:559]: logging the perf issues here https://github.com/gitcoinco/web/issues/6729

a few more things i want to do... 
- add to subject line: upcoming dates, num notifications, num announcements, etc
- what else?? will know more as this ships.

Comment (owocki) [reply app/retail/emails.py:559]: grants

File: app/retail/templates/shared/activity.html [hunk 3]
```diff
@@ -312,12 +312,17 @@
...
+            <img class="fa-icon-png" src="{% static "v2/images/emails/link-regular.png" %}">
+          </a>
           {% endif %}
 
+
           <div class="dropdown-menu dropdown-menu-right shadow font-smaller-4 px-0">
             {% if can_pin %}
-              <a class="pin_activity mr-2 dropdown-item px-3 font-smaller-5" data-toggle="tooltip" title="Pinned post will display at the top of the feed." href=#
-              data-url="{{row.url}}"
+              <a class="pin_activity mr-2 dropdown-item px-3 font-smaller-5" data-toggle="tooltip" title="Pinned post will display at the top of the feed." hrefo=#
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html:324]: hrefo ?

File: app/retail/templates/shared/activity.html [hunk 4]
```diff
@@ -312,12 +312,17 @@
...
+              data-url="{{rw.url}}"
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html:325]: rw ? :P

File: app/retail/templates/shared/activity.html [hunk 5]
```diff
@@ -287,7 +287,7 @@
             >
               <span class="action {% if row.metadata.liked %}open{%endif%}">
                 {% if for_email %}
-                  <img class="fa-icon-png" src=""{% static "v2/images/emails/heart-regular.png" %}">
+                  <img class="fa-icon-png" src="{{base_url}}static/v2/images/emails/heart-regular.png">
```
Comment (thelostone-mc) [inline app/retail/templates/shared/activity.html:290]: Did `{% static` not work ?

Comment (thelostone-mc) [reply app/retail/templates/shared/activity.html:290]: @owocki

File: app/retail/templates/emails/new_bounty.html [hunk 6]
```diff
@@ -152,7 +196,7 @@ <h3>{% trans "Matching your profile" %}</h3>
 {% endif %}
 
 <p>
-  {% trans "You are receiving this email because your email address is on the notification list" %}.  <a href="{% url 'email_settings' subscriber.priv %}" style="text-decoration: underline;">{% trans "Manage Email Settings" %}</a>.
+  {% trans "You are receiving this email bc you are subscribed to daily emails." %}  <a href="{% url 'email_settings' subscriber.priv %}" style="text-decoration: underline;">{% trans "Manage Email Settings" %}</a> | <a href="{% url 'email_settings' subscriber.priv %}?type=new_bounty_notifications" style="text-decoration: underline;">{% trans "Unsubscribe" %}</a>.
```
Comment (thelostone-mc) [inline app/retail/templates/emails/new_bounty.html:199]: bc ?

**Pull request merged**